src/App.tsx

```diff
@@ -1,14 +1,71 @@
 /* eslint-disable max-len */
-import React from 'react';
+import React, { useEffect, useMemo, useState } from 'react';
 import 'bulma/css/bulma.css';
 import '@fortawesome/fontawesome-free/css/all.css';
 
 import { TodoList } from './components/TodoList';
 import { TodoFilter } from './components/TodoFilter';
-import { TodoModal } from './components/TodoModal';
 import { Loader } from './components/Loader';
+import { getTodos } from './api';
+import { Todo } from './types/Todo';
+import { TodoModal } from './components/TodoModal';
 
 export const App: React.FC = () => {
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [userId, setUserId] = useState<number>(0);
+
+  const [selectedTodo, setSelectedTodo] = useState<Todo | null>(null);
+
+  const [filter, setFilter] = useState('all');
+  const [query, setQuery] = useState('');
+
+  const newModel = (newTodo: Todo) => {
+    setUserId(newTodo.userId);
+
+    return setSelectedTodo(newTodo);
+  };
+
+  useEffect(() => {
+    getTodos()
+      .then(data => {
+        setTodos(data);
+      })
+      .catch(error => {
+        throw new Error(error);
+      });
+  }, []);
+
+  const reset = () => {
+    setUserId(0);
+    setSelectedTodo(null);
+  };
+
+  const filteredTodosByOption = useMemo(() => {
+    return todos.filter((todoToFilter) => {
+      switch (filter) {
+        case 'all':
+          return true;
```

Comment:
so, if filter is 'all' then you just do useless iteration? It's a wasting of a memory. Better do filter in every case